.github/workflows/ci.yml

```diff
@@ -6,31 +6,21 @@ on:
   pull_request:
   workflow_dispatch:
 jobs:
-  static-site-lint:
-    runs-on: ubuntu-latest
-    defaults:
-      run:
-        working-directory: static-site/
-    steps:
-      - uses: actions/checkout@v4
-      - uses: actions/setup-node@v4
-        with:
-          node-version: '16'
-      - run: npm ci
-      - run: npm run lint
-
   test:
     if: github.repository == 'nextstrain/nextstrain.org'
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-node@v4
         with:
-          node-version: '16'
+          node-version-file: 'package.json'
       - run: node --version
       - run: npm ci
```

Comment:
[not related to this line, just using it to start a thread]

We are getting a lot of transient test failures (as well as some legitimate ones!). Currently the failure is related to tests which hit GitHub endpoints (and unrelated to any changes in this PR):

```console
FAIL test/get_available.test.js
  ● getAvailable community URLs correctly interpret the GitHub branch › Explicit non-default branch

    expect(received).toEqual(expected) // deep equality

    Expected: 1
    Received: 0

      44 |       }
      45 |       const branches = new Set(requests.map((r) => parseBranchFromAddress(r)));
    > 46 |       expect(branches.size).toEqual(1);
         |                             ^
      47 |       expect(branches).toContain(testCase.branch);
      48 |     });
      49 |   }

      at Object.<anonymous> (test/get_available.test.js:46:29)
```

```console
FAIL test/auspice_client_requests.test.js
  ● smoke testing URLs described in auspice_client_requests.json › Check getSourceInfo API for a community (GitHub) URL

    expect(received).toEqual(expected) // deep equality

    Expected: 200
    Received: 404

      13 |       const res = await fetch(url, {redirect: 'manual'});
      14 |
    > 15 |       expect(res.status).toEqual(testCase.expectStatusCode || 200);
         |                          ^
      16 |
      17 |       if (testCase.responseIsJson) await testResponseIsJson(res);
      18 |

      at Object.<anonymous> (test/auspice_client_requests.test.js:15:26)

```

My guess (haven't looked into it) is that tests are running without any GitHub credentials and we're running up against rate limits. It's not the remit of this PR, but going forward we should try hard to make our test suites less prone to stochastic failures like these.

Reply:
> My guess (haven't looked into it) is that tests are running without any GitHub credentials and we're running up against rate limits.

Seems likely. Rate limit for unauthenticated requests is [60 per hour per IP](https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#primary-rate-limit-for-unauthenticated-users). Searching for `prefix=/community` in the code shows that there's 12 requests per test run, so 5 back-to-back test runs will do it.

I wonder if it's as easy as using the [default `GITHUB_TOKEN`](https://docs.github.com/en/actions/security-guides/automatic-token-authentication):

```yaml
- run: npm run test:ci
  env:
    GITHUB_TOKEN: ${{ github.token }}
```

Reply:
> Searching for prefix=/community in the code shows that there's 12 requests per test run

If you look at the server logs after a single test run (e.g. `npm run test:ci` then look at `test/server.log`) we make 127 GET requests to https://api.github.com, but some of them are cache hits and so (maybe?) aren't counted in the limits.

Reply:
[Looks promising](https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#primary-rate-limit-for-github_token-in-github-actions):

> You can use the built-in `GITHUB_TOKEN` to authenticate requests in GitHub Actions workflows.
> The rate limit for `GITHUB_TOKEN` is 1,000 requests per hour per repository

I'll make a separate PR.

Reply:
Oh I haven't looked at server.log, that's useful.

I see 85 GET requests. Without any `GITHUB_TOKEN`, I ran `npm run test:ci` 4 times in a row successfully, then the amount of failures varied:

1. all passed
2. all passed
3. all passed
4. all passed
5. 4 failed
6. 16 failed
7. 4 failed
8. 4 failed

Not sure what's going on, but using the GitHub Actions provided token should be an improvement: #817